platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeAnnotatedMethodReader.java

```diff
@@ -238,9 +322,17 @@ private static Type[] toArray(Type owner, List<ParameterInfo> parameters) {
     }
 
     private static String getPropertyName(ExecutableElement method) {
-        String methodName = method.getSimpleName().toString();
-        String property = methodName.startsWith("get") ? methodName.substring(3) : methodName;
-        return Character.toLowerCase(property.charAt(0)) + property.substring(1);
+        return getPropertyName(method.getSimpleName().toString());
+    }
+
+    private static String getPropertyName(String methodName) {
+        if (methodName.startsWith("is") && methodName.length() > 2 && Character.isUpperCase(methodName.charAt(2))) {
+            return Character.toLowerCase(methodName.charAt(2)) + methodName.substring(3);
```

Comment:
:x: This is definitely covered by `PropertyAccessorType` already.

Reply:
The problem is, that `PropertyAccessorType` is in `model-core` and we don't want to depend on that. So we either move that class of `model-core` or have some duplicated code.

We don't want to depend on `model-core` since it depends on `core-api` and we want to upgrade some properties in `core-api`.